### PR TITLE
Late block reorgs metrics and logs

### PR DESCRIPTION
**Pull request by potuz**
This PR introduces metrics and logs related to late block reorgs. It logs when an attempted reorg was aborted at 10 and 12 seconds due to attestations. 

It also removes the metric `missedPayloadIDFilledCount` that was only incremented in the case where the block arrives late

### CODE REVIEW

**Review by terencechain — COMMENTED** [review]
This PR looks logically correct, but something smells that I can't put into words.

From the comment and the name `shouldOverrideFCU`, it sounds like `return true` implies that a reorg is happening. But it's the opposite `shouldOverrideFCU` `return false` is reorg is happening. As a caller, if I can `shouldOverrideFCU` and get false, I would think that I don't need to do anything else. I think the word `Override` can be interpreted both ways and will be confusing

I would welcome a better name or boolean structure. Maybe something like `canReorgBlock`

File: beacon-chain/blockchain/forkchoice_update_execution.go [hunk 1]
```diff
@@ -84,3 +84,41 @@ func (s *Service) forkchoiceUpdateWithExecution(ctx context.Context, newHeadRoot
...
+	if proposingSlot == currentSlot {
+		proposerHead := s.ForkChoicer().GetProposerHead()
+		if proposerHead != newHeadRoot {
+			return true
+		}
+		log.WithFields(logrus.Fields{
+			"root":   fmt.Sprintf("%#x", newHeadRoot),
+			"weight": headWeight,
+		}).Info("Attempted late block reorg aborted due to attestations at 12 seconds")
+		lateBlockFailedAttemptFirstThreshold.Inc()
```
Comment (terencechain) [inline beacon-chain/blockchain/forkchoice_update_execution.go:105]: Shouldn't 12s be the second attempt? It's our final bet right

File: beacon-chain/blockchain/forkchoice_update_execution.go [hunk 2]
```diff
@@ -84,3 +84,41 @@ func (s *Service) forkchoiceUpdateWithExecution(ctx context.Context, newHeadRoot
...
+		if err != nil {
+			log.WithError(err).Error("could not compute seconds since slot start")
+		}
+		if secs >= doublylinkedtree.ProcessAttestationsThreshold {
+			log.WithFields(logrus.Fields{
+				"root":   fmt.Sprintf("%#x", newHeadRoot),
+				"weight": headWeight,
+			}).Info("Attempted late block reorg aborted due to attestations at 10 seconds")
+			lateBlockFailedAttemptSecondThreshold.Inc()
+		}
```
Comment (terencechain) [inline beacon-chain/blockchain/forkchoice_update_execution.go:121]: It'll be confusing if a block arrives later than 10s and also trigger this condition

**Comment by potuz** [comment]
> This PR looks logically correct, but something smells that I can't put into words.
> 
> From the comment and the name `shouldOverrideFCU`, it sounds like `return true` implies that a reorg is happening. But it's the opposite `shouldOverrideFCU` `return false` is reorg is happening. As a caller, if I can `shouldOverrideFCU` and get false, I would think that I don't need to do anything else. I think the word `Override` can be interpreted both ways and will be confusing
> 
Nope, it's exactly the opposite. `shouldOverrideFCU == true` implies that the reorg is happening

File: beacon-chain/blockchain/forkchoice_update_execution.go (hunk 1 above)
Comment (potuz) [reply beacon-chain/blockchain/forkchoice_update_execution.go:105]: yep got those two mixed up, thanks

**Review by terencechain — DISMISSED** [review]

**Pull request merged**